From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: John Johansen <john.johansen@canonical.com>
Date: Thu, 24 Aug 2023 12:38:01 -0700
Subject: [PATCH 71/78] UBUNTU: SAUCE: apparmor4.0.0 [74/82]: advertise
 disconnected.path is available

BugLink: https://bugs.launchpad.net/bugs/2028253

While disconnected.path has been available for a while it was never
properly advertised as a feature. Fix this so that userspace doesn't
need special casing to handle it.

Signed-off-by: John Johansen <john.johansen@canonical.com>
(cherry picked from https://gitlab.com/jjohansen/apparmor-kernel)
Signed-off-by: Andrea Righi <andrea.righi@canonical.com>
---
 security/apparmor/apparmorfs.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/security/apparmor/apparmorfs.c b/security/apparmor/apparmorfs.c
index 6323f72d10ec..52332fdf180e 100644
--- a/security/apparmor/apparmorfs.c
+++ b/security/apparmor/apparmorfs.c
@@ -2603,6 +2603,7 @@ static struct aa_sfs_entry aa_sfs_entry_unconfined[] = {
 
 static struct aa_sfs_entry aa_sfs_entry_profile[] = {
 	AA_SFS_FILE_BOOLEAN("interruptible",		1),
+	AA_SFS_FILE_BOOLEAN("disconnected.path",	1),
 	{ }
 };
 
